pkg/alertmanager/alertmanager.go

```diff
@@ -191,7 +192,7 @@ type Replicator interface {
 }
 
 // New creates a new Alertmanager.
-func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
+func New(cfg *Config, reg *prometheus.Registry, tmplExternalURL *url.URL) (*Alertmanager, error) {
```

Comment:
If we're already passing the URL in `ApplyConfig()`, we can skip adding a new parameter here. On the initial sync ([right after the Alertmanager is created](https://github.com/grafana/mimir/blob/f52911d917c8c52e0da6a59348a64dd7f7622072/pkg/alertmanager/multitenant.go#L907-L929)), the AM will get the correct URL to be used when testing receivers.

I'd remove this new parameter.